### DISCUSSION TITLE
Working with third-party APIs

### DISCUSSION THREAD

**Discussion by man0so1** [def-studio/telegraph#558] [Q&A]
Hello again! I have a problem and I need your help. My bot should receive data from the user (there is no problem with this) and send it to my API for further actions, and accordingly receive a response from this API. I'm trying to do this through the built-in Laravel `HTTP` facade with the parameters I need. I'm testing everything using `ngrok`, in case he might be involved in this problem. If I execute this request through the `artisan` test command, it will work quickly and return the data I need. But if I do this through the bot's command, it will take an infinitely long time to receive a response and eventually give an error about waiting too long. And also using the `ngrok` request checker, which is located on the `4040` port, I noticed that this request is constantly replicating itself in my handler, although the user did not make a request for it. This request is constantly being executed with a wait of almost a minute, but still without receiving a response. I'm asking for tips on how I should work with the API through my bot. I will be very grateful!

**Comment by fabio-ivona** [comment]
Hi, I guess there is an unhandled exception on your webhook controller that may lead to the request replication, can you check?
